zerver/models.py

```diff
@@ -1766,6 +1775,8 @@ def can_subscribe_other_users(self) -> bool:
         return self.has_permission("invite_to_stream_policy")
 
     def can_invite_others_to_realm(self) -> bool:
+        if self.realm.invite_to_realm_policy == Realm.POLICY_NOBODY:
+            return False
```

Comment:
I moved this check to `has_permission`, since it really should be part of the common infrastructure even though only this setting makes nobody a possible value.